src/commonMain/kotlin/exchange.dydx.abacus/calculator/MarginCalculator.kt

```diff
@@ -263,21 +260,34 @@ internal object MarginCalculator {
     }
 
     /**
-     * @description Determine if collateral should be transferred out for an isolated margin trade
+     * @description Determine if collateral should be transferred out from child subaccount for an isolated margin trade
      */
     fun getShouldTransferOutCollateral(
         parser: ParserProtocol,
         subaccount: Map<String, Any>?,
         tradeInput: Map<String, Any>?,
     ): Boolean {
-        val isDecreasingPositionSize = !getIsIncreasingPositionSize(parser, subaccount, tradeInput)
+        val isPositionFullyClosed = getIsPositionFullyClosed(parser, subaccount, tradeInput)
```

Comment:
wouldn't we still want to check is decreasingPositionSize as well? This was to not transfer if they flipped sides but their net position was decreasing.

Reply:
resolved offline -- this is a new function that was added for the diff state and fixed logic is the right one!